packages/wrapper/src/editorAppClassic.ts

```diff
@@ -79,18 +79,10 @@ export class EditorAppClassic extends EditorAppBase {
         }
 
         if (this.config.editorOptions?.['semanticHighlighting.enabled'] !== undefined) {
-            let userConfig;
-            try {
-                userConfig = JSON.parse(await getUserConfiguration());
-            } catch (e) {
-                userConfig = {};
-            }
-            const json = JSON.stringify({
-                ...userConfig,
-                'editor.semanticHighlighting.enabled': this.config.editorOptions['semanticHighlighting.enabled']
-            });
-            await updateUserConfiguration(json);
+            const editorConfig = workspace.getConfiguration('editor.semanticHighlighting');
+            await editorConfig.update('enabled', this.config.editorOptions['semanticHighlighting.enabled'], ConfigurationTarget.Workspace);
```

Comment:
You now update the workspace configuration instead of the user configuration, is it in purpose?

Note that you can also chose to update the memory configuration (there are many layers of configuration that are merged)

Reply:
Is `ConfigurationTarget.WorkspaceFolder` (3) = user configuration ?

Reply:
You are setting the configuration via the Extension API while I was suggesting to set it via the service directly. Via the api you don't have access to all the targets

Reply:
But the service does not expose those functions or does it?

Reply:
What do you mean? you can use
```typescript
import { ConfigurationTarget, IConfigurationService, StandaloneServices } from 'vscode/services'

StandaloneServices.get(IConfigurationService).updateValue('editor.semanticHighlighting', 'enabled', ConfigurationTarget.MEMORY)
```

Reply:
Thank you, I was wondering if it is exposed via `@codingame/monaco-vscode-configuration-service-override`.

Reply:
Done ⬇️

Reply:
So you've chosen USER over MEMORY? :)

Reply:
Yes, I thought it makes sense. It is anyway only for the classical config. These days, I would recommend to use the extended mode directly. It could make sense to drop the other mode at some point in the future.

Reply:
There is only one example left that uses the `classic` / monarch app config.

Reply:
I was thinking... you can provide a default configuration to the initialize method also, see https://github.com/CodinGame/monaco-vscode-api/blob/61752f5a8e77b1359c8648f8455d048d483c7e79/demo/src/setup.common.ts#L245-L247

Reply:
Generally a good idea, but not for this specific problem, because it is just a workaround. I want to introduce the view service here (derived from your demo) soon. I started that work on branch https://github.com/TypeFox/monaco-languageclient/tree/oct-vscode-web and may come back to that idea then. WDYT?

Reply:
Why not for this pecific problem? at least you have the guarantee to not impact anything outside the current session

Reply:
Ok, that became a bigger change, but hopefully for good. I moved the `userConfiguration` out of the editor app to the renamed `vscodeApiConfig`. If semanticHighlighting is now configured in the editorOptions this is now set to workspaceConfig.configurationDefaults.

Can you tell me why this is a such? https://github.com/TypeFox/monaco-languageclient/blob/env-views-options/packages/wrapper/test/wrapper.test.ts#L244-L245

It works as expected when I check the Langium Classic example, but I didn't expect `configuredByThem` as return in the test. I thought it is just true.

Reply:
I don't know :thinking: maybe you check it too soon?

Reply:
@CGNonofr a timeout works. Now the test ensure a worker factory is available, too. That prevented prevents some errors. The test didn't fail, but there where unexpected errors.